impala/_rpc/hiveserver2.py

```diff
@@ -226,10 +224,46 @@ def get_result_schema(service, operation_handle):
         else:
             schema.append((name, type_, None, None, None, None, None))
 
-
     return schema
 
 
+_re_nulls = re.compile(b'^(\x00)+$')
+
+
+if six.PY3:
+
+    def _get_null(values, nulls, i):
+        if not isinstance(nulls, bytes):
+            nulls = nulls.encode('utf8')
+
+        # i // 8 is the byte, i % 8 is position in the byte; get the
+        # int repr and pull out the bit at the corresponding pos
+        is_null = False
+        # Ref HUE-2722, HiveServer2 sometimes does not add not put
+        # trailing '\x00'.
+        if len(values) != len(nulls):
+            nulls = nulls + (b'\x00' * (len(values) - len(nulls)))
+        # Hive encodes nulls differently than Impala
+        # (\x00 vs \x00\x00 ...)
+        if not _re_nulls.match(nulls):
+            is_null = bool(nulls[i // 8]) & (1 << (i % 8))
```

Comment:
The `bool` seems suspect to me here.  Doesn't this destroy all the info in the byte you're pulling out?